dali/operators/reader/loader/webdataset_loader.h

```diff
@@ -117,6 +117,10 @@ class DLL_PUBLIC WebdatasetLoader : public Loader<CPUBackend, vector<Tensor<CPUB
   size_t sample_index_ = 0;
   FileStream::MappingReserver mmap_reserver_;
   std::once_flag multiple_files_single_component;
+
+  bool is_index_generated_ = true;
```

Comment:
I don't like this name. It reads like "has the index been generated?" and it should read as "is the index generated on the fly". Alternative:
```suggestion
  bool generate_index_ = true;
```
This tells us that we should generate the index (and the variable is indeed used like this in webdataset_loader.cc:352.

Alternatively, you can flip this flag's meaning:
```suggestion
  bool use_index_file_ = false;
```